meson.build

```diff
@@ -41,6 +41,10 @@ if get_option('readline-provider') == 'readline'
 	if readline.found()
 		add_project_arguments('-DHAVE_READLINE', language: 'c')
 	endif
+
+  if cc.has_function('rl_replace_line', prefix: '#include <readline/readline.h>', args: ['-lreadline'])
```

Comment:
I think we can replace `args: ['-lreadline']` with `dependencies: [readline]`, this would prevent the detection from failing if the pkg-config readline file points to something else than `/usr/lib`.